src/nvim/os/env.c

```diff
@@ -135,7 +135,20 @@ int os_setenv(const char *name, const char *value, int overwrite)
   }
 #endif
   uv_mutex_lock(&mutex);
-  int r = uv_os_setenv(name, value);
+  int r;
+#ifdef WIN32
+  // LC_ALL, LANG, LANGUAGE and LC_MESSAGES must be obtained using getenv()
+  // in libintl, so we must use _putenv_s().
```

Comment:
This is rather unfortunate. We are almost going in circles, with environment management. 

Would it make sense to do *both*? I.e. `_putenv_s` _and_ `uv_os_setenv`.

Reply:
> Would it make sense to do both? I.e. _putenv_s and uv_os_setenv.

I do not think so. This problem occurs because the value written by `SetEnvironmentVariable()` cannot be read by `getenv()`. But the values ​​set with `putenv()`, `_putenv()` and `_putenv_s()` can be obtained with `GetEnvironmentVariable()`. So calling both will be useless.

I think there are other following solutions to this problem:

- Use `_wputenv_s()` and `_wgetenv_s()` instead of `uv_os_getenv()` and `uv_os_setenv()`
- Apply a patch to the library(`libintl` etc) that uses `getenv()`, Change to use `GetEnvironmentVariable()`.
- I don't know if I can do it, but we rewrite `IAT` and replace `getenv()` itself with a function provided by neovim.